src/main/java/net/plavcak/jenkins/plugins/scmskip/SCMSkipTools.java

```diff
@@ -93,16 +93,20 @@ private static boolean inspectChangeSet(List<ChangeLogSet<?>> changeLogSets, SCM
 
         ChangeLogSet.Entry matchedEntry = null;
 
+        boolean allSkipped = true;
+        
         for (Object entry : changeLogSet.getItems()) {
             if (entry instanceof ChangeLogSet.Entry && inspectChangeSetEntry((Entry) entry, matcher)) {
                 matchedEntry = (Entry) entry;
+            } else {
+                allSkipped = false;
                 break;
             }
         }
 
         String commitMessage  = combineChangeLogMessages(changeLogSet);
 
-        if (matchedEntry == null) {
+        if (allSkipped) {
```

Comment:
Shouldn't this be:
`if (!allSkipped)` because, here we know that there is at least one entry in the `changeLogSet `which does _not_ match on the skip pattern and we need to build.

Reply:
Negative logic here just makes it a little obscure.  If ANY of the changeset messages contain [skip ci] (skip pattern) , then allSkipped = false and the job is triggered.  If ALL of the changeset messages contain [skip ci], then allSkipped = true (it nevers gets updated) and the job is  not triggered.

case 1:
* [skip ci] 
* foo
* bar

run job

case 2:

* [skip ci]
* [skip ci]
* [skip ci]

don't run job.

Reply:
You could simply use a single line java regex ensure only the first line of a concatenated commit message drives the skip/no skip decide (as indicated by ThirteenThirtySeven).  You lose the multiline commit message support in the process from previous release.  It's messy.  You can have multiple line support or single line (for merge request trigger) but not both without something like this PR.